### ISSUE TITLE
Dockerfile adjustments

### ISSUE THREAD

**Issue by SuperCoolCucumber** [IINemo/lm-polygraph#206]
1) The path to `requirements.txt` is incorrect in the Dockerfile -- it is located in the main directory of the project.
2) The `app` directory and the `CMD ["polygraph_server"]` command are only required if one desires to run frontend. I did not use the frontend app so I skipped those. It might make sense to make a separate Dockerfile for those who only intend to use the methods of the framework to avoid installing extra packages.
3) There is no `jupyter` in the requirements which is needed to run the demo notebooks.
4) CUDA drivers are not pulled by default in the Dockerfile but I assume this depends on the specific hardware configurations. I have used `nvcr.io/nvidia/pytorch:24.05-py3` image to use CUDA on my cluster.

**Comment by rvashurin** [comment]
@SuperCoolCucumber thanks a lot for your feedback! 

Indeed, existing Dockerfile is only suitable for running the demo app, not for the standalone library usage. We will work on adding library-specific Dockerfile in the near future, and will update you when this is being worked on.

**Comment by rvashurin** [comment]
@SuperCoolCucumber Can you elaborate a little bit how you were planning to interact with Polygraph in a docker? Run benchmarks from CLI, example notebooks, or you just need an encapsulated environment where library lives and can run, so you can exec into running container and do whatever?

**Comment by SuperCoolCucumber** [comment]
Yeah, I'm using a cluster so I do need and encapsulated environment to use and interact with the library. I've used Polygraph to compute certain uncertainty estimations for my models on my data. I've introduced some changes into model inference and ran the code for some whitebox methods.
